packages/js/components/src/experimental-select-control/select-control.tsx

```diff
@@ -103,20 +114,51 @@ export const SelectControl = ( {
 
 					break;
 				case useCombobox.stateChangeTypes.InputKeyDownEnter:
+				case useCombobox.stateChangeTypes.FunctionSelectItem:
 				case useCombobox.stateChangeTypes.ItemClick:
 				case useCombobox.stateChangeTypes.InputBlur:
 					if ( selectedItem ) {
 						onSelect( selectedItem );
-						setInputValue(
-							multiple ? '' : itemToString( selectedItem )
-						);
+						if ( clearSearchOnSelect ) {
+							onInputChange( '' );
+							setInputValue( '' );
```

Comment:
Optionally clear the input value upon selection.
Not sure if we want to separate `InputBlur` from this, and do clear in for that? I guess play around with it and see what feels natural.